src/js/controllers/publish.js

```diff
@@ -1,9 +1,42 @@
 'use strict';
 
-angular.module('starter.controllers').controller('publishController', function($scope, $cordovaImagePicker, $cordovaCamera, $ionicPlatform, $state) {
+angular.module('starter.controllers').controller('publishController', function($scope, $cordovaImagePicker, $cordovaCamera, $ionicPlatform, $state, $ionicPopup, $translate) {
 
   $scope.images = [];
 
+  $scope.categoryPopup = function() {
+    $scope.data = {};
+
+    var myPopup = $ionicPopup.show({
+      title: 'Choose a Category',
+      scope: $scope,
+      cssClass: 'popup-vertical-buttons',
+      buttons: [
+        {
+          text: $translate.instant('publish_adopt'),
+          type: 'button-full button-positive',
+          onTap: function(e) {
+            return $scope.data.category;
```

Comment:
Actually, we don't know how that will save the parameter just yet. So I left it like that.